test/python/transpiler/test_permrowcol.py

```diff
@@ -592,6 +656,81 @@ def test_add_cnot_never_adds_hadamard_gates_to_a_bidirectional_circuit(self):
 
         self.assertEqual(h_gates, 0)
 
+    def test_add_cnot_adds_corresponding_row_operations_on_parity_matrix(self):
+        coupling_list = [(0, 1), (0, 3), (1, 2), (1, 4), (2, 5), (3, 4), (4, 5)]
+        coupling = CouplingMap(coupling_list)
+        permrowcol = PermRowCol(coupling)
+        parity_mat = np.array(
+            [
+                [0, 1, 0, 1, 1, 0],
+                [1, 1, 1, 1, 1, 0],
+                [1, 0, 0, 0, 1, 1],
+                [1, 1, 1, 0, 1, 0],
+                [1, 0, 1, 0, 1, 0],
+                [1, 0, 1, 0, 1, 1],
+            ]
+        )
+        target = 1
+        control = 0
+        circ = QuantumCircuit(6)
+
+        permrowcol._add_cnot(circ, parity_mat, control, target)
+        #     correct_permutation_matrix = np.array(
+        #         [
+        #             [0, 0, 0, 1, 0, 0],
+        #             [0, 0, 1, 0, 0, 0],
+        #             [0, 0, 0, 0, 0, 1],
+        #             [0, 1, 0, 0, 0, 0],
+        #             [0, 0, 0, 0, 1, 0],
+        #             [1, 0, 0, 0, 0, 0],
+        #         ]
+        #     )
+
+        #     self.assertTrue(len(circ.data) == 1)
+        self.assertEqual(sum(parity_mat[:, 3]), 1)
+
+        self.assertEqual(parity_mat[0, 3], 1)
+
+    def test_perm_row_col_returns_valid_output_with_a_common_case(self):
```

Comment:
This can be broadened later to use different backends, but all in all it's great that you got this working.